ambari-server/src/main/java/org/apache/ambari/server/upgrade/UpgradeCatalog300.java

```diff
@@ -542,6 +580,65 @@ protected void addAmbariConfigurationTable() throws SQLException {
     dbAccessor.addPKConstraint(AMBARI_CONFIGURATION_TABLE, "PK_ambari_configuration", AMBARI_CONFIGURATION_CATEGORY_NAME_COLUMN, AMBARI_CONFIGURATION_PROPERTY_NAME_COLUMN);
   }
 
+  /**
+   * Creates new tables for changed kerberos data.
+   *
+   * @throws SQLException
+   */
+  protected void upgradeKerberosTables() throws SQLException {
+    boolean keytabPrincipalTableCreated = false;
+    boolean mappingTableCreated = false;
+
+    if (!dbAccessor.tableExists(KERBEROS_KEYTAB_TABLE)) {
```

Comment:
I believe that `dbAccessor.createTable` and `dbAccessor.addPKContraint` are safe to call without checking for this. Each ensure they are no re-adding the requested construct.   I don't think it hurts to do this though.